src/main/java/bio/terra/service/common/gcs/GcsUriUtils.java

```diff
@@ -0,0 +1,49 @@
+package bio.terra.service.common.gcs;
+
+import com.google.cloud.storage.BlobId;
+
+public final class GcsUriUtils {
+  private static final String GS_BUCKET_PATTERN = "[a-z0-9_.\\-]{3,222}";
+
+  /**
+   * Parse a Google Cloud Storage URI into its component pieces
+   *
+   * @param uri of type gs://<bucket_name>/<file_path_inside_bucket>
+   * @return Object representing uri pieces
+   */
+  public static BlobId parseBlobUri(String uri) {
+    BlobId blobId = BlobId.fromGsUtilUri(uri);
+    validateBlobUri(blobId, uri);
+    return blobId;
+  }
+
+  public static void validateBlobUri(String uri) throws IllegalArgumentException {
+    validateBlobUri(BlobId.fromGsUtilUri(uri), uri);
```

Comment:
This is identical to `parseBlobUri()` except that this doesn't return the `BlobId`. If we want to have both, this should call the other to avoid code duplication.
```suggestion
    parseBlobUri(uri);
```

Reply:
Not sure I agree with this, I think it's more important that the functions are named according to their purpose. 
Having Validate call Parse call Validate is more confusing than just duplicating that one line of code.

Reply:
Users can still call `validate()` instead of `parse()`; they don't need to know that the implementation of validate internally calls parse, which calls another validate method. But I agree that it's not much code. It's up to you if you want to keep it the way it is or change it.